### ISSUE TITLE
[feature] Loud Defederation

### ISSUE THREAD

**Issue by SoniEx2** [superseriousbusiness/gotosocial#1933]
### Is your feature request related to a problem ?

Sometimes you need a better solution than quiet-quitting facebook.

### Describe the solution you'd like.

Loud defederation, like silent defederation, involves serving different responses to the same object IDs based on what's requesting them. But with loud defederation you feed them full objects instead of error messages or connection resets.

### Describe alternatives you've considered.

N/A

**Comment by igalic** [comment]
> But with loud defederation you feed them full objects instead of error messages or connection resets.

so how is it defederated then?
what are those full objects?

**Comment by SoniEx2** [comment]
They can be really creative.

For users, you can feed them the most generic profile you can come up with.

For posts, replacing the whole post with "This instance has been blocked due to having a Facebook problem" would be rather effective.

**Comment by trysdyn** [comment]
This is the kind of thing that starts half-understanding rumors that a specific codebase is malicious. Please be very careful proceeding here.

On a more fundamental level: I've fantasized about and drafted a plan for sending Wildebeest/P92 to a hellpit where I feed it fake objects or even do bad things to its database, but I feel this kind of activity belongs in a sidecar service or clever terminator tricks, not in the actual service itself.

There's a level of complexity here where I feel just providing a checkbox for "Send litter objects to this instance because you *really* dislike them" will do more than the average self-hoster will understand (caching, target-to-non-target federation of context, etc) and I'm struggling to find a way to present this functionality in a way that will make them understanding of the effects before they click it.

I also worry even if this kind of thing is limited to targeting P92, there will be a ripple effect that'll distribute the "litter" objects to other instances via context federation. It all depends on how P92 handles federation and we're frankly likely never going to know for sure.

To put it simply: even for this situation where I'm pretty fine with being quite rude to P92, I don't know how much I like the idea of packaging a feature where we're essentially deliberately causing a fediverse split-horizon regarding what our posts/profiles actually are.

I would prefer we figure out how to fingerprint P92 and just refuse to serve it on a code level if we're going to do anything.

**Comment by SoniEx2** [comment]
most of the worries about it poisoning other instances can be solved by only doing it with authorized fetch.

but the core of this feature request is simply "let admins customize the defederation response". between sending a 403, or sending alternative object contents, or whatever one might want.

you don't need to *actively* send them the "litter" objects or mess with their databases.

**Comment by tsmethurst** [comment]
This seems like a bad idea:

1. Sending some object is more expensive than just immediately aborting the request with 403 like we do now.
2. It could be used really badly by assholes.
3. It will make people distrust messages from GtS instances.

**Comment by SoniEx2** [comment]
We're aware of how this could be used to attempt to gaslight your own users. We also don't think it's gonna be a serious problem in practice. Because nobody's ever done it before.

And a WAF or even plain old firewall or fail2ban would be way better than relying on GtS to handle malicious instances trying to cause issues with (1). It should still send 403 (or is it 401?) for fully-unauthenticated requests.

**Comment by tsmethurst** [comment]
I'm gonna close this. In addition to the points I already raised, I'd also add:

1. It doesn't add any benefit for GtS users/admins, it's purely negative for other servers (which seems like a bad route to start going down).
2. It's more stuff to maintain, when we have plenty of stuff on our plate as-is.

**Comment by SoniEx2** [comment]
Being negative for other servers isn't inherently a bad thing. Defederation is negative for other servers, and yet we have it.
